netdata-installer.sh

```diff
@@ -469,32 +465,75 @@ bundle_libmosquitto() {
+                     "${NETDATA_LOCAL_TARBALL_OVERRIDE_MOSQUITTO}"
+  then
+    if run tar -xf "${tmp}/${MOSQUITTO_PACKAGE_BASENAME}" -C "${tmp}" && \
+       build_libmosquitto "${tmp}/mosquitto-${MOSQUITTO_PACKAGE_VERSION}" && \
+       copy_libmosquitto "${tmp}/mosquitto-${MOSQUITTO_PACKAGE_VERSION}" && \
+       rm -rf "${tmp}"
+    then
+      run_ok "libmosquitto built and prepared."
+    else
+      run_failed "Failed to build libmosquitto. The install process will continue, but you will not be able to connect this node to Netdata Cloud."
+    fi
   else
-    progress "Using provided mosquitto tarball ${NETDATA_LOCAL_TARBALL_OVERRIDE_MOSQUITTO}"
-    run cp "${NETDATA_LOCAL_TARBALL_OVERRIDE_MOSQUITTO}" "${tmp}/${MOSQUITTO_PACKAGE_BASENAME}"
+    run_failed "Unable to fetch sources for libmosquitto. The install process will continue, but you will not be able to connect this node to Netdata Cloud."
   fi
+}
 
-  if [ ! -f "${tmp}/${MOSQUITTO_PACKAGE_BASENAME}" ] || [ ! -s "${tmp}/${MOSQUITTO_PACKAGE_BASENAME}" ]; then
-    run_failed "unable to find a usable libmosquitto source archive, Netdata Cloud will not be available"
-    return 0
-  fi
+bundle_libmosquitto
 
-  grep "${MOSQUITTO_PACKAGE_BASENAME}\$" "${INSTALLER_DIR}/packaging/mosquitto.checksums" > "${tmp}/sha256sums.txt" 2> /dev/null
+# -----------------------------------------------------------------------------
 
-  # Checksum validation
-  if ! (cd "${tmp}" && safe_sha256sum -c "sha256sums.txt"); then
-    run_failed "mosquitto files checksum validation failed."
-    return 0
-  fi
+build_libwebsockets() {
+  cmake "${1}"
+  make -C "${1}"
+}
+
+copy_libwebsockets() {
+  target_dir="${PWD}/externaldeps/libwebsockets"
 
-  run tar -xf "${tmp}/${MOSQUITTO_PACKAGE_BASENAME}" -C "${tmp}"
+  run mkdir -p "${target_dir}"
 
-  build_libmosquitto "${tmp}/mosquitto-${MOSQUITTO_PACKAGE_VERSION}" && copy_libmosquitto "${tmp}/mosquitto-${MOSQUITTO_PACKAGE_VERSION}" && rm "${tmp}"
+  run cp "${1}/lib/libwebsockets.a" "${target_dir}/libwebsockets.a"
+  run cp -r "${1}/include" "${target_dir}"
 }
 
-bundle_libmosquitto
+bundle_libwebsockets() {
+  if [ -n "${SKIP_LIB_WEBSOCKETS}" ] || pkg-config "libwebsockets >= 3" ; then
+    return 0
+  fi
+
+  progress "Prepare libwebsockets"
+
+  LIBWEBSOCKETS_PACKAGE_VERSION="$(cat packaging/libwebsockets.version)"
+
+  tmp="$(mktemp -d -t netdata-libwebsockets-XXXXXX)"
+  LIBWEBSOCKETS_PACKAGE_BASENAME="${LIBWEBSOCKETS_PACKAGE_VERSION}.tar.gz"
+
+  if fetch_and_verify "websockets" \
+                      "https://github.com/warmcat/libwebsockets/archive/${LIBWEBSOCKETS_PACKAGE_BASENAME}" \
+                      "${LIBWEBSOCKETS_PACKAGE_BASENAME}" \
+                      "${tmp}" \
+                      "${NETDATA_LOCAL_TARBALL_OVERRIDE_LIBWEBSOCKETS}"
+  then
+    if run tar -xf "${tmp}/${LIBWEBSOCKETS_PACKAGE_BASENAME}" -C "${tmp}" && \
+       build_libwebsockets "${tmp}/websockets-${LIBWEBSOCKETS_PACKAGE_VERSION}" && \
```

Comment:
On my system it unpacks to `${tmp}/libwebsockets-3.2.2` not `${tmp}/libwebsockets-v3.2.2`